resources/images/controller/images.go

```diff
@@ -234,16 +235,37 @@ func (s *SoftwareImagesController) NewImage(w rest.ResponseWriter, r *rest.Reque
 	case ErrModelArtifactNotUnique:
 		l.Error(err.Error())
 		s.view.RenderError(w, r, cause, http.StatusUnprocessableEntity, l)
+	case ErrModelParsingArtifactFailed:
+		l.Error(err.Error())
+		s.view.RenderError(w, r, formatArtifactUploadError(err), http.StatusBadRequest, l)
 	case ErrModelMissingInputMetadata, ErrModelMissingInputArtifact,
 		ErrModelInvalidMetadata, ErrModelMultipartUploadMsgMalformed,
-		ErrModelArtifactFileTooLarge, ErrModelParsingArtifactFailed:
+		ErrModelArtifactFileTooLarge:
 		l.Error(err.Error())
 		s.view.RenderError(w, r, cause, http.StatusBadRequest, l)
 	}
 
 	return
 }
 
+func formatArtifactUploadError(err error) error {
+	// remove generic message
+	errMsg := strings.TrimSuffix(err.Error(), ": "+ErrModelParsingArtifactFailed.Error())
+
+	// handle specific cases
+
+	if strings.Contains(errMsg, "invalid checksum") {
+		return errors.New(errMsg[strings.Index(errMsg, "invalid checksum"):])
+	}
+
+	if strings.Contains(errMsg, "unsupported version") {
+		return errors.New(errMsg[strings.Index(errMsg, "unsupported version"):] +
+			"; supported versions are: 1, 2, 3")
```

Comment:
Only 1 and 2 are supported ATM. 3 is WIP.

Reply:
fixed